pkg/scheduler/framework/plugins/podtopologyspread/filtering_test.go

```diff
@@ -1450,12 +1534,66 @@ func TestSingleConstraint(t *testing.T) {
 				"node-y": framework.Unschedulable,
 			},
 		},
+		{
+			name: "pods spread across nodes as 2/2, maxSkew is 2, and the number of domains < minDomains, then no node fits",
+			pod: st.MakePod().Name("p").Label("foo", "").SpreadConstraint(
+				2,
+				"node",
+				v1.DoNotSchedule,
+				st.MakeLabelSelector().Exists("foo").Obj(),
+				pointer.Int32(3), // larger than the number of domains(2)
+			).Obj(),
+			nodes: []*v1.Node{
+				st.MakeNode().Name("node-a").Label("node", "node-a").Obj(),
+				st.MakeNode().Name("node-b").Label("node", "node-b").Obj(),
+			},
+			existingPods: []*v1.Pod{
+				st.MakePod().Name("p-a1").Node("node-a").Label("foo", "").Obj(),
+				st.MakePod().Name("p-a2").Node("node-a").Label("foo", "").Obj(),
+				st.MakePod().Name("p-b1").Node("node-b").Label("foo", "").Obj(),
+				st.MakePod().Name("p-b2").Node("node-b").Label("foo", "").Obj(),
+			},
+			enableMinDomains: true,
+			wantStatusCode: map[string]framework.Code{
+				"node-a": framework.Unschedulable,
+				"node-b": framework.Unschedulable,
+			},
+		},
+		{
+			name: "pods spread across nodes as 2/2/1, maxSkew is 2, and the number of domains < minDomains, then the third node fits",
+			pod: st.MakePod().Name("p").Label("foo", "").SpreadConstraint(
+				2,
+				"node",
+				v1.DoNotSchedule,
+				st.MakeLabelSelector().Exists("foo").Obj(),
+				pointer.Int32(4), // larger than the number of domains(3)
```

Comment:
Can we also cover:

- minDomains < # of domains
- also, test ScheduleAnyway

Reply:
> minDomains < # of domains

Did you add this one already?

Reply:
👍 
I add the test case for `minDomains < # of domains` for https://github.com/kubernetes/kubernetes/pull/108362/commits/eea5c3b105d55041d3471b4328945911d4a74c77#r823730962

I didn't add the test for `minDomains < # of domains`. in previous commit 
This was because:
- MinDomains is 1 when `MinDomains` is nil, so `minDomains < # of domains` is true in all other test cases.
- `Filter` method doesn't change its behavior depending on the value of `MinDomains`